include/boost/spirit/home/x3/core/action.hpp

```diff
@@ -82,14 +81,12 @@ namespace boost { namespace spirit { namespace x3
             typedef typename
                 traits::attribute_of<action<Subject, Action>, Context>::type
             attribute_type;
-            typedef traits::make_attribute<attribute_type, unused_type> make_attribute;
             typedef traits::transform_attribute<
-                typename make_attribute::type, attribute_type, parser_id>
+                unused_type, attribute_type, parser_id>
             transform;
 
             // synthesize the attribute since one is not supplied
-            typename make_attribute::type made_attr = make_attribute::call(unused_type());
-            typename transform::type attr = transform::pre(made_attr);
+            typename transform::type attr = transform::pre(unused_type());
```

Comment:
I think we can replace this with `attribute_type attr{};`. No one should specialize something like `transform_attribute<unused_type, int>`, it would be nonsense.